src/L1/rollup/ScrollChain.sol

```diff
@@ -405,29 +563,28 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
         }
         BatchHeaderV0Codec.storeVersion(batchPtr, _version);
 
+        //EnforcedBatchParameters memory parameters = enforcedBatchParameters;
         (bytes32 _parentBatchHash, uint256 _batchIndex, uint256 _totalL1MessagesPoppedOverall) = _beforeCommitBatch(
             _parentBatchHeader,
-            _chunks
+            _chunks,
+            0
         );
         BatchHeaderV0Codec.storeBatchIndex(batchPtr, _batchIndex);
 
         // versions 2 and 3 both use ChunkCodecV1
         (bytes32 _dataHash, uint256 _totalL1MessagesPoppedInBatch) = _commitChunksV1(
             _totalL1MessagesPoppedOverall,
             _chunks,
-            _skippedL1MessageBitmap
+            0 // parameters.maxInclusionDelay
```

Comment:
is this a TODO? 
since we also call `_commitBatchWithBlobProof` from `commitAndFinalizeBatch` we apply the same logic when in permissionless mode. Just to double check: This means that if there's lots of L1 messages then a permissionless batch will need to include all L1 messages first before any other tx created on L2 (via permissionless batch production toolkit)?

Reply:
yes, it's a todo. fixed